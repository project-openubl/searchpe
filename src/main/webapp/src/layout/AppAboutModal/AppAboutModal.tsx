import React from "react";
import {
  AboutModal,
  Flex,
  FlexItem,
  List,
  ListItem,
  TextContent,
} from "@patternfly/react-core";
import {
  GithubIcon,
  GlobeIcon,
  EnvelopeIcon,
  InfoAltIcon,
} from "@patternfly/react-icons";

import brandImage from "images/logo-navbar.svg";

export interface AppAboutModalProps {
  isOpen: boolean;
  onClose: () => void;
}

export const AppAboutModal: React.FC<AppAboutModalProps> = ({
  isOpen,
  onClose,
}) => {
  return (
    <AboutModal
      isOpen={isOpen}
      onClose={onClose}
      trademark="COPYRIGHT © 2020."
      brandImageSrc={brandImage}
      brandImageAlt="Logo"
      productName=""
    >
      <TextContent className="pf-u-py-xl">
        <h4>About</h4>
        <p>
          <a
            href="https://project-openubl.github.io/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Searchpe
          </a>{" "}
          allows you to consume the data exposed from SUNAT through the 'padrón
          reducido' available online.
        </p>
      </TextContent>
      <TextContent>
        <h4>Links</h4>
        <List>
          <ListItem>
            <a
              href="https://project-openubl.github.io/"
              target="_blank"
              rel="noopener noreferrer"
            >
              Website
            </a>
          </ListItem>
          <ListItem>
            <a
              href="https://project-openubl.github.io/"
              target="_blank"
              rel="noopener noreferrer"
            >
              Documentation
            </a>
          </ListItem>
        </List>
      </TextContent>
      <div className="pf-c-about-modal-box__strapline">
        <Flex>
          <FlexItem>
            <a
              href="https://github.com/project-openubl/searchpe"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i>
                <GithubIcon />
              </i>{" "}
              Source
            </a>
          </FlexItem>
          <FlexItem>
            <a
              href="https://projectopenubl.zulipchat.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i>
                <GlobeIcon />
              </i>{" "}
              Discussion forum
            </a>
          </FlexItem>
          <FlexItem>
            <a href="mailto:projectopenubl+subscribe@googlegroups.com?subject=Subscribe&body=Hello">
              <i>
                <EnvelopeIcon />
              </i>{" "}
              Mailing list
            </a>
          </FlexItem>
          <FlexItem>
            <a
              href="https://github.com/project-openubl/searchpe/issues"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i>
                <InfoAltIcon />
              </i>{" "}
              Issue tracking
            </a>
          </FlexItem>
        </Flex>
      </div>
    </AboutModal>
  );
};
