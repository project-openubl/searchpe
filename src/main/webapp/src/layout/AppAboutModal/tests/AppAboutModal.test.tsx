import React from "react";
import { shallow } from "enzyme";
import { AppAboutModal } from "../AppAboutModal";

it("AppAboutModal", () => {
  const wrapper = shallow(<AppAboutModal isOpen={true} onClose={jest.fn()} />);
  expect(wrapper).toMatchSnapshot();
});
