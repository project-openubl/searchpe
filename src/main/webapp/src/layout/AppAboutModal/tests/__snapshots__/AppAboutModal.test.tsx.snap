// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppAboutModal 1`] = `
<AboutModal
  appendTo={null}
  backgroundImageSrc=""
  brandImageAlt="Logo"
  brandImageSrc="logo-navbar.svg"
  className=""
  isOpen={true}
  noAboutModalBoxContentContainer={false}
  onClose={[MockFunction]}
  productName=""
  trademark="COPYRIGHT © 2020."
>
  <TextContent
    className="pf-u-py-xl"
  >
    <h4>
      About
    </h4>
    <p>
      <a
        href="https://project-openubl.github.io/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Searchpe
      </a>
       
      allows you to consume the data exposed from SUNAT through the 'padrón reducido' available online.
    </p>
  </TextContent>
  <TextContent>
    <h4>
      Links
    </h4>
    <List>
      <ListItem>
        <a
          href="https://project-openubl.github.io/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Website
        </a>
      </ListItem>
      <ListItem>
        <a
          href="https://project-openubl.github.io/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Documentation
        </a>
      </ListItem>
    </List>
  </TextContent>
  <div
    className="pf-c-about-modal-box__strapline"
  >
    <Flex>
      <FlexItem>
        <a
          href="https://github.com/project-openubl/searchpe"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i>
            <GithubIcon
              color="currentColor"
              noVerticalAlign={false}
              size="sm"
            />
          </i>
           
          Source
        </a>
      </FlexItem>
      <FlexItem>
        <a
          href="https://projectopenubl.zulipchat.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i>
            <GlobeIcon
              color="currentColor"
              noVerticalAlign={false}
              size="sm"
            />
          </i>
           
          Discussion forum
        </a>
      </FlexItem>
      <FlexItem>
        <a
          href="mailto:projectopenubl+subscribe@googlegroups.com?subject=Subscribe&body=Hello"
        >
          <i>
            <EnvelopeIcon
              color="currentColor"
              noVerticalAlign={false}
              size="sm"
            />
          </i>
           
          Mailing list
        </a>
      </FlexItem>
      <FlexItem>
        <a
          href="https://github.com/project-openubl/searchpe/issues"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i>
            <InfoAltIcon
              color="currentColor"
              noVerticalAlign={false}
              size="sm"
            />
          </i>
           
          Issue tracking
        </a>
      </FlexItem>
    </Flex>
  </div>
</AboutModal>
`;
